ddl/ddl_worker.go

```diff
@@ -537,24 +549,9 @@ func (w *worker) handleDDLJobQueue(d *ddlCtx) error {
 			d.mu.hook.OnJobRunBefore(job)
 			d.mu.RUnlock()
 
-			// Meta releated txn default is DiskFullOpt_AllowedOnAlmostFull to support
-			// all the ddl job queue operations or other meta change. But we only want
-			// to support the Drop Table like ddls to be executed when TiKV is disk full.
-			switch job.Type {
-			case model.ActionDropSchema, model.ActionDropTable, model.ActionDropIndex, model.ActionTruncateTable, model.ActionDropTablePartition, model.ActionDropView, model.ActionDropSequence, model.ActionDropIndexes, model.ActionTruncateTablePartition:
-				txn.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlmostFull)
-			default:
-				txn.SetDiskFullOpt(kvrpcpb.DiskFullOpt_NotAllowedOnFull)
-			}
-
 			// If running job meets error, we will save this error in job Error
 			// and retry later if the job is not cancelled.
 			schemaVer, runJobErr = w.runDDLJob(d, t, job)
-			if job.IsCancelled() {
```

Comment:
Why do you remove it?

Reply:
I want to judge it before check the dependency.
To let Job be able to canceled when TiKV is disk full.

Reply:
It may be canceled in runDDLJob

Reply:
Right!
DDLs in normal queue is not blocked, so, there is no dependency problem for the canceling operation on DDLs in index queue. And there is also no need to cancel DDLs in normal queue.